### PR TITLE
Revert "feat(core): revert support for parallel after stages (#2194)"

### PR DESCRIPTION
**Pull request by danielpeach**
Do not merge.

This reverts commit f99de659b08ae9afe3b0e0b66c88a2d578e075e8.

Was reverted because the parent stage never completed, but I haven't been able to reproduce this. Would appreciate a look over to see if there is an obvious situation that I'm not thinking of that would produce the bug.

### CODE REVIEW

File: orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/ContinueParentStageHandler.kt [hunk 1]
```diff
@@ -46,14 +43,23 @@ class ContinueParentStageHandler(
...
-        when {
-          stage.hasTasks() -> stage.runFirstTask()
-          else -> queue.push(CompleteStage(stage))
+      if (message.phase == STAGE_BEFORE) {
+        if (stage.allBeforeStagesComplete()) {
+          when {
+            stage.hasTasks() -> stage.runFirstTask()
+            else             -> queue.push(CompleteStage(stage))
+          }
+        } else if (!stage.anyBeforeStagesFailed()) {
```
Comment (lwander) [inline orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/ContinueParentStageHandler.kt]: What if this is true? Do we have to handle the failure here?

File: orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/ContinueParentStageHandler.kt [hunk 2]
```diff
@@ -63,18 +69,7 @@ class ContinueParentStageHandler(
     if (firstTask.status == NOT_STARTED) {
       queue.push(StartTask(this, firstTask))
     } else {
-      log.warn("Ignoring $messageType for ${id} as tasks are already running")
-    }
-  }
-
-  private fun Stage.runAfterStages() {
```
Comment (lwander) [inline orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/ContinueParentStageHandler.kt]: Why was this deleted?

File: orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/ContinueParentStageHandler.kt (hunk 1 above)
Comment (danielpeach) [reply orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/ContinueParentStageHandler.kt]: I think the `ContinueParentStage` message is only sent in the case where a synthetic stage doesn't fail (the failing synthetic stage would end up completing the execution).

I think this handles the case where a `ContinueParentStage` message is sent -> a sibling synthetic stage fails -> this handler picks up the `ContinueParentStage` message and should do nothing because the execution is going to be completed by a different handler.

File: orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/ContinueParentStageHandler.kt (hunk 2 above)
Comment (danielpeach) [reply orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/ContinueParentStageHandler.kt]: Don't know, this was in the original revert. It's not used.

**Comment by duftler** [comment]
@ajordens @robfletcher Any objections to merging this one?

**Comment by ajordens** [comment]
Merge it and we'll see what happens.
